js_modules/dagster-ui/packages/ui-core/src/app/analytics.tsx

```diff
@@ -1,5 +1,11 @@
-import {createContext, useCallback, useContext, useEffect, useMemo} from 'react';
+import {createContext, useCallback, useContext, useLayoutEffect} from 'react';
 import {useLocation, useRouteMatch} from 'react-router-dom';
+import {atom, useRecoilValue, useSetRecoilState} from 'recoil';
+
+export const currentPageAtom = atom<{path: string; specificPath: string}>({
```

Comment:
I used a recoil atom here to avoid having yet another provider and having to order them properly (which in some cases necessitates another provider yay). Cloud already uses Recoil

Reply:
I'm not quite sure I follow how Recoil avoids another provider / ordering problem, this PR adds a RecoilProvider?

Reply:
The Recoil provider is at the root, that's pretty much the extent of ordering that we need to think about.